core/tracker/incldelay.go

```diff
@@ -31,20 +31,33 @@ func NewInclDelayFunc(eth2Cl eth2wrap.Client, dutiesFunc dutiesFunc) func(contex
 // newInclDelayFunc extends NewInclDelayFunc with abstracted callback.
 func newInclDelayFunc(eth2Cl eth2wrap.Client, dutiesFunc dutiesFunc, callback func([]int64)) func(context.Context, core.Slot) error {
 	// dutyStartSlot is the first slot we can instrument (since dutiesFunc will not have duties from older slots).
-	var dutyStartSlot int64
-	var dssMutex sync.Mutex
+	var (
+		dutyStartSlot int64
+		dssMutex      sync.Mutex
+	)
 
-	return func(ctx context.Context, current core.Slot) error {
+	setStartSlot := func(slot int64) {
 		dssMutex.Lock()
 		defer dssMutex.Unlock()
 
+		dutyStartSlot = slot
+	}
+
+	getStartSlot := func() int64 {
+		dssMutex.Lock()
+		defer dssMutex.Unlock()
+
+		return dutyStartSlot
+	}
+
+	return func(ctx context.Context, current core.Slot) error {
 		// blockSlot the block we want to instrument.
 		blockSlot := current.Slot - inclDelayLag
 
-		if dutyStartSlot == 0 {
-			dutyStartSlot = current.Slot // Set start slot.
+		if getStartSlot() == 0 {
```

Comment:
What happens if in the span of time between `getStartSlot()` and `setStartSlot()`, `dutyStartSlot` gets changed by another function call?

Is this something that can happen?

Do we care if this happens?

Reply:
Technically it cannot happen as dutyStartSlot is set when charon is started i.e., when dutyStartSlot is 0.

Reply:
Or we can say it is set when we receive first slot tick event since charon start.

Reply:
Hmmm, we call `tracker.NewInclDelayFunc(eth2Cl, sched.GetDutyDefinition)` in app.go. So if we call `app.Run` multiple times then it will create new set of dutyStartSlot variable for each `app.Run` call.